### PR TITLE
Add support for hetzner dns console

### PR DESCRIPTION
**Pull request by schreibubi**
Add support for the hetzner dns console

### CODE REVIEW

File: ddclient.in [hunk 1]
```diff
@@ -5725,6 +5738,163 @@ sub nic_cloudflare_update {
...
+  fully.qualified.host         ## the host registered with the service.
+
+Example ${program}.conf file entries:
+  ## single host update using an API token
+  protocol=hetzner,                                            \\
+  zone=dns.zone,                                               \\
+  password=hetzner-api-token                                   \\
+  myhost.com
+
+  ## multiple host update
```
Comment (SuperSandro2000) [inline ddclient.in:5766]:
```suggestion
```

File: ddclient.in [hunk 2]
```diff
@@ -5725,6 +5738,163 @@ sub nic_cloudflare_update {
...
+        for my $domain (@hosts) {
+            (my $hostname = $domain) =~ s/\.$config{$key}{zone}$//;
+            my $ipv4  = delete $config{$domain}{'wantipv4'};
+            my $ipv6  = delete $config{$domain}{'wantipv6'};
+
+            info("getting Hetzner Zone ID for %s", $domain);
+
+            # Get zone ID
+            my $url = "https://$config{$key}{'server'}/zones?";
+            $url .= "name=" . $config{$key}{'zone'};
```
Comment (SuperSandro2000) [inline ddclient.in:5801]:
```suggestion
            my $url = "https://$config{$key}{'server'}/zones?name=" . $config{$key}{'zone'};
```

File: ddclient.in [hunk 3]
```diff
@@ -5725,6 +5738,163 @@ sub nic_cloudflare_update {
...
+######################################################################
+sub nic_hetzner_examples {
+    return <<"EoEXAMPLE";
+o 'hetzner'
+
+The 'hetzner' protocol is used by DNS service offered by www.hetzner.com.
+
+Configuration variables applicable to the 'hetzner' protocol are:
+  protocol=hetzner             ##
+  server=fqdn.of.service       ## defaults to dns.hetzner.com/api/v1
```
Comment (SuperSandro2000) [inline ddclient.in:5755]:
```suggestion
  protocol=hetzner 
  server=dns.hetzner.com/api/v1  ## can be omited 
```
I don't think anyone would want to change that.

**Comment by schreibubi** [comment]
@SuperSandro2000 Addressed your suggested changes, rebased to resolve merge conflicts

**Pull request merged**